mobile_wallet/src/lib.rs

```diff
@@ -349,6 +325,98 @@ fn sign_message_aux(input: &str) -> anyhow::Result<String> {
     ))?)
 }
 
+#[derive(common::SerdeDeserialize)]
+struct MicroCCDAmount {
+    #[serde(rename = "microGtuAmount")]
+    amount: u64,
+}
+
+#[derive(common::SerdeDeserialize)]
+struct AccountAddressJSON {
+    address: String,
+}
+
+#[derive(common::SerdeDeserialize)]
+#[serde(untagged)]
+enum JSONPayload {
+    UpdateContract {
+        amount:       MicroCCDAmount,
```

Comment:
Actually we already have a struct which is exactly what you need here I think. And has sensible JSON
```rust
UpdateContract {
    #[serde(flatten)]
    payload: base::UpdateContractPayload
}
```

Reply:
Yeah, except that the parameter/message fields have different names, but that might be more of a JS/me issue :thinking:

Reply:
The correct thing would be to change the field name to message in javascript, so we are aligned. I guess.